lighthouse-core/config/config.js

```diff
@@ -752,63 +738,54 @@ class Config {
     }
 
     const coreList = Runner.getGathererList();
-    passes.forEach(pass => {
-      pass.gatherers.forEach(gathererDefn => {
-        if (!gathererDefn.instance) {
-          let GathererClass = gathererDefn.implementation;
-          if (!GathererClass) {
-            // See if the gatherer is a Lighthouse core gatherer
-            const name = gathererDefn.path;
-            const coreGatherer = coreList.find(a => a === `${name}.js`);
-
-            let requirePath = `../gather/gatherers/${name}`;
-            if (!coreGatherer) {
-              // Otherwise, attempt to find it elsewhere. This throws if not found.
-              requirePath = Runner.resolvePlugin(name, configPath, 'gatherer');
-            }
-
-            GathererClass = require(requirePath);
+    const fullPasses = passes.map(pass => {
+      const gathererDefns = Config.expandGathererShorthand(pass.gatherers).map(gathererDefn => {
```

Comment:
this has become quite the inner loop, WDYT about extracting a method for the actual `requireing` part as it was before?

Reply:
> WDYT about extracting a method for the actual requireing part as it was before?

ok, attempted. LMKWYT